src/App.tsx

```diff
@@ -15,47 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+function getSorted(goods: string[], sortField: string, reverse: boolean) {
+  const preperedGood = [...goods];
+
+  if (sortField) {
+    preperedGood.sort((good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.Alphabetically:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preperedGood.reverse();
+  }
+
+  return preperedGood;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getSorted(goodsFromServer, sortField, reverse);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${cn({
+            'is-light': sortField !== SortType.Alphabetically,
+          })}`}
+          onClick={() => setSortField(SortType.Alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${cn({
+            'is-light': sortField !== SortType.Length,
+          })}`}
+          onClick={() => setSortField(SortType.Length)}
+
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${cn({
+            'is-light': reverse === false,
+          })}`}
+          onClick={() => setReverse(!reverse)}
```

Comment:
```suggestion
          onClick={() => setReverse((prevReverse) => !prevReverse)}
```

Reply:
It is safer to pass stateUpdater callback if new state value is based on previous one